src/GPS.cpp

```diff
@@ -0,0 +1,73 @@
+#include "GPS.hpp"
+#include <nmea0183/GPS.hpp>
+
+using namespace nmea0183;
+using namespace marnav;
+using namespace std;
+using namespace gps_base;
+
+GPS::GPS(Driver& driver)
+    : m_driver(driver)
+{
+}
+
+gps_base::GPS_SOLUTION_TYPES GPS::getPositionType(
+    marnav::utils::optional<marnav::nmea::mode_indicator> mode)
+{
+    // TODO: check returns
+    switch (*mode) {
+        case marnav::nmea::mode_indicator::invalid:
+            return GPS_SOLUTION_TYPES::INVALID;
+        case marnav::nmea::mode_indicator::autonomous:
+            return GPS_SOLUTION_TYPES::AUTONOMOUS_2D;
+        case marnav::nmea::mode_indicator::differential:
+            return GPS_SOLUTION_TYPES::DIFFERENTIAL;
+        case marnav::nmea::mode_indicator::estimated:
+            return GPS_SOLUTION_TYPES::INVALID;
+        case marnav::nmea::mode_indicator::manual_input:
+            return GPS_SOLUTION_TYPES::INVALID;
+        case marnav::nmea::mode_indicator::simulated:
+            return GPS_SOLUTION_TYPES::INVALID;
+        case marnav::nmea::mode_indicator::data_not_valid:
+            return GPS_SOLUTION_TYPES::NO_SOLUTION;
+        case marnav::nmea::mode_indicator::precise:
+            return GPS_SOLUTION_TYPES::INVALID;
+        default:
+            return GPS_SOLUTION_TYPES::INVALID;
+    }
+}
+
+gps_base::Position nmea0183::GPS::getPosition(marnav::nmea::rmc const& rmc,
+    marnav::nmea::gsa const& gsa)
+{
+    gps_base::Position position;
+    // TODO: time now or message time?
```

Comment:
I think it should be message time, as you are reporting the position from this message, not doing any processing that may change this information.

Reply:
We had this discussion a time ago. I believe that the time should be the timestamp in the moment that the message was created. The gps_base::Position in this case